packages/date-picker/src/DatePicker/DatePicker.spec.tsx

```diff
@@ -1,33 +1,64 @@
-import { act, render, screen } from "@testing-library/react"
-import React from "react"
+import { act, fireEvent, render, screen, waitFor } from "@testing-library/react"
+import React, { useState } from "react"
+import { FieldMessageStatus } from "@kaizen/draft-form"
 import userEvent from "@testing-library/user-event"
-import { DatePicker } from "./DatePicker"
+import { DatePicker, ValidationResponse } from "./DatePicker"
 import "@testing-library/jest-dom"
+import { DatePickerProps } from "."
 
 const defaultProps = {
   id: "date-picker",
   labelText: "Choose date",
   selectedDay: undefined,
   initialMonth: new Date(2022, 2),
   onDayChange: jest.fn(),
+  onValidate: jest.fn(),
+}
+
+const MockDatePicker = ({
+  selectedDay,
+  ...restProps
+}: Partial<DatePickerProps>) => {
+  const [status, setStatus] = useState<FieldMessageStatus>("default")
+  const [validationMessage, setValidationMessage] = useState<string>()
+  const [selectedDate, setValueDate] = useState<Date | undefined>(selectedDay)
+
+  const handleValidation = (validationResponse: ValidationResponse) => {
+    validationResponse.status && setStatus(validationResponse.status)
+    validationResponse.validationMessage &&
+      setValidationMessage(validationResponse.validationMessage)
+  }
+  return (
+    <DatePicker
+      {...defaultProps}
+      {...restProps}
+      onValidate={handleValidation}
+      onDayChange={setValueDate}
+      status={status}
+      validationMessage={validationMessage}
+      selectedDay={selectedDate}
+    />
+  )
 }
 
 describe("<DatePicker />", () => {
   it("renders DatePicker with an empty input value", async () => {
-    render(<DatePicker {...defaultProps} />)
+    render(<MockDatePicker {...defaultProps} />)
 
     expect(screen.getByRole("combobox")).toHaveValue("")
   })
 
   it("renders DatePicker and displays inital date within input", async () => {
-    render(<DatePicker {...defaultProps} selectedDay={new Date(2022, 2, 1)} />)
+    render(
+      <MockDatePicker {...defaultProps} selectedDay={new Date(2022, 2, 1)} />
+    )
 
     // Make sure date renders in the button
     expect(screen.getByDisplayValue("Mar 1, 2022")).toBeInTheDocument()
   })
 
   it("renders DatePicker and shows/hides calendar on button press", async () => {
```

Comment:
```suggestion
  it("shows/hides calendar on button press", async () => {
```

And similar for other descriptions. The outer `describe` already tells us it's about the DatePicker.

Also unsure which button is being referred to in these descriptions (and in the test for that matter) since there are so many buttons in a datepicker. Worth revisiting these tests later to make them a little clearer.